SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -425,6 +427,36 @@ FHandoverChangeState USpatialActorChannel::CreateInitialHandoverChangeState(cons
 	return HandoverChanged;
 }
 
+void USpatialActorChannel::UpdateVisibleComponent(AActor* InActor)
+{
+	// Make sure that the Actor is not always relevant.
+	if (InActor->bAlwaysRelevant)
+	{
+		return;
+	}
+
+	// Make sure that the InActor is not a PlayerController, GameplayDebuggerCategoryReplicator or GameMode.
+	if (InActor->IsA(APlayerController::StaticClass()) || InActor->IsA(AGameplayDebuggerCategoryReplicator::StaticClass())
+		|| InActor->IsA(AGameModeBase::StaticClass()))
+	{
+		return;
+	}
+
+	// Unreal applies the following rules (in order) in determining the relevant set of Actors for a player:
+	// If the Actor is hidden (bHidden == true) and the root component does not collide then the Actor is not relevant.
+	// We apply the same rules to add/remove the Visible component to an actor that determines if clients will checkout the actor or
+	// not.
+	if (InActor->IsHidden() && (!InActor->GetRootComponent() || !InActor->GetRootComponent()->IsCollisionEnabled()))
```

Comment:
Or we just change this to have `&& !bAlwaysRelevant` and the if statement below `|| bAlwaysRelevant`. What do you think?

Reply:
yep thats a good solution I will add the bAlwaysRelevant check here.